### PR TITLE
backend: Add /sitemap endpoint

### PR DESCRIPTION
**Pull request by barthalion**

### CODE REVIEW

**Comment by barthalion** [comment]
This is now ready. Once it's merged, I will alias `/sitemap.txt` to the new endpoint, and add the following robots.txt directly to HTTP server config:

```
User-agent: *
Allow: /
Host: https://beta.flathub.org
Sitemap: https://beta.flathub.org/sitemap.txt
```

**Comment by razzeee** [comment]
But we don't want beta to be indexed?

File: backend/app/sitemap.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        "/apps/collection/recently-updated",
+        "/badges",
+        "/feeds",
+        "/privacy-policy",
+        "/statistics",
+        "/terms-and-conditions",
+    ]
+
+    applications = [f"/apps/details/{appid}" for appid in apps.list_appstream()]
+    projectgroups = [
```
Comment (razzeee) [inline backend/app/sitemap.py:26]: Missing developer pages? https://beta.flathub.org/apps/collection/developer/Bilal%20Elmoussaoui

File: backend/app/sitemap.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        "/apps/collection/popular",
+        "/apps/collection/recently-updated",
+        "/badges",
+        "/feeds",
+        "/privacy-policy",
+        "/statistics",
+        "/terms-and-conditions",
+    ]
+
+    applications = [f"/apps/details/{appid}" for appid in apps.list_appstream()]
```
Comment (razzeee) [inline backend/app/sitemap.py:25]: Categories? https://beta.flathub.org/apps/category/AudioVideo

File: backend/app/sitemap.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        "/apps/collection/editors-choice-apps",
+        "/apps/collection/editors-choice-apps",
+        "/apps/collection/editors-choice-games",
+        "/apps/collection/popular",
+        "/apps/collection/recently-updated",
+        "/badges",
+        "/feeds",
+        "/privacy-policy",
+        "/statistics",
+        "/terms-and-conditions",
```
Comment (razzeee) [inline backend/app/sitemap.py:22]: https://beta.flathub.org/languages

**Comment by razzeee** [comment]
This also should handle other languages

**Comment by razzeee** [comment]
You could parse https://github.com/flathub/website/blob/main/frontend/next-i18next.config.js I guess? Or just copy that (and keep it up ta date)

> It already is, though… I will add robots.txt to disallow crawlers. We're going to need this endpoint eventually anyway.
image
Might be old?

**Comment by barthalion** [comment]
There are some results at https://www.google.com/search?hl=en&q=site%3Abeta.flathub.org so I don't think it fully works.

**Comment by razzeee** [comment]
Seems like we could remove these with the google webmaster tools https://www.voog.com/blog/how-to-remove-your-website-or-web-page-from-google

File: backend/app/sitemap.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    redis_key = "sitemap_text"
+
+    if sitemap := db.redis_conn.get(redis_key):
+        return sitemap
+
+    pages = [
+        "/about",
+        "/apps",
+        "/apps/collection/editors-choice-apps",
+        "/apps/collection/editors-choice-apps",
```
Comment (razzeee) [inline backend/app/sitemap.py:14]: This is duplicated

**Comment by razzeee** [comment]
https://github.com/flathub/website/pull/387 is an extended version of this

**Pull request closed without merging**